spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,194 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe(Tapioca::Compilers::Dsl::ActiveRecordTypedStore) do
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecordTypedStore classes") do
+      expect(subject.processable_constants).to(be_empty)
+    end
+
+    it("gather only TypedStore classes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+        class Shop < ActiveRecord::Base
+        end
+
+        class User
+        end
+      RUBY
+
+      expect(constants_from(content)).to(eq(["Post"]))
+    end
+  end
+
+  describe("#decorate") do
+    def rbi_for(content)
+      with_content(content) do
+        parlour = Parlour::RbiGenerator.new(sort_namespaces: true)
+        subject.decorate(parlour.root, Post)
+        parlour.rbi
+      end
+    end
+
+    it("generate RBI for TypedStore classes without accessor") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadatado do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+
+      RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for TypedStore classes with accessor") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: true)
```

Comment:
Why do we have `blank: false` here? If it does not affect the test, let's remove it.

Reply:
```suggestion
            s.string(:reviewer)
```